timezone_manager.py

```diff
@@ -36,32 +44,30 @@ def init():
             logger.warning(
                 f"user config and system timezones confilct, updating system config to {MeticulousConfig[CONFIG_USER][TIME_ZONE]}"
             )
-            status = TimezoneManager.set_system_timezone(
-                MeticulousConfig[CONFIG_USER][TIME_ZONE]
-            )
-            # If fails, set the system_timezone as the user timezone and report the error
-            if status != "SUCCESS":
+            try:
+                TimezoneManager.set_system_timezone(
+                    MeticulousConfig[CONFIG_USER][TIME_ZONE]
+                )
+            except TimezoneManagerError:
+                # If fails, set the system_timezone as the user timezone and report the error
                 logger.error(
                     f"failed to set system TZ, updating user TZ to {TimezoneManager.__system_timezone} "
                 )
                 MeticulousConfig[CONFIG_USER][
                     TIME_ZONE
                 ] = TimezoneManager.__system_timezone
-            MeticulousConfig.save()
+            finally:
+                MeticulousConfig.save()
 
     @staticmethod
-    def update_timezone(new_timezone: str) -> str:
-
-        if MeticulousConfig[CONFIG_USER][TIME_ZONE] != new_timezone:
-            SUCCESS = TimezoneManager.set_system_timezone(new_timezone)
-            logger.debug(f"update timezone status: {SUCCESS}")
-            if SUCCESS:
-                MeticulousConfig[CONFIG_USER][TIME_ZONE] = new_timezone.rstrip(
-                    '"'
-                ).lstrip('"')
-                MeticulousConfig.save()
-            return SUCCESS
-        return "Success"
+    def update_timezone(new_timezone: str) -> None:
+        stripped_new_tz = new_timezone.rstrip('"').lstrip('"')
+        if MeticulousConfig[CONFIG_USER][TIME_ZONE] != stripped_new_tz:
+            try:
+                TimezoneManager.set_system_timezone(stripped_new_tz)
+                logger.debug("update timezone status: Success")
+            except TimezoneManagerError as e:
+                raise TimezoneManagerError(f"Error updating timezone:\n\t{e}", log=True)
```

Comment:
Nit: This format string looks a bit weird in sentry. Nothing wrong with it per-se tho